### ISSUE TITLE
aws_ec2_managed_prefix_list --> replaced --> delete-failed

### ISSUE THREAD

**Issue by BornToBeRoot** [hashicorp/terraform-provider-aws#20991]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

Terraform v1.0.3
on windows_amd64

### Affected Resource(s)

* aws_ec2_managed_prefix_list

### Terraform Configuration Files

```
resource "aws_ec2_managed_prefix_list" "xxx" {
  name           = "${var.environment}-pl-xxx"
  address_family = "IPv4"
  max_entries    = length(var.xxx)

  dynamic "entry" {
    for_each = var.xxx

    content {
      cidr        = entry.value.ip
      description = entry.value.description
    }
  }
}
```

Assign the pl to security groups, routing table with terraform....

### Panic Output
```
  # module.aws_pl.aws_ec2_managed_prefix_list.xxxxxxxxxxxxx must be replaced
-/+ resource "aws_ec2_managed_prefix_list" "xxxxxxxxxxxxx" {
      ~ arn            = "arn:aws:ec2:eu-central-1:xxxxxxxxxxxxx:prefix-list/pl-xxxxxxxxxxxxx" -> (known after apply)
      ~ id             = "pl-xxxxxxxxxxxxx" -> (known after apply)
      ~ max_entries    = 2 -> 3 # forces replacement
        name           = "dev-pl-xxxxxxxxxxxxx"
      ~ owner_id       = "xxxxxxxxxxxxx" -> (known after apply)
        tags           = {
            "Name" = "dev-pl-xxxxxxxxxxxxx"
        }
      ~ version        = 1 -> (known after apply)
        # (2 unchanged attributes hidden)

      + entry {
          + cidr        = "10.x.x.x/16"
          + description = "x"
        }
        # (2 unchanged blocks hidden)
    }
```
```
Error: error waiting for EC2 Managed Prefix List (pl-xxxxxxxxxxxxx) deletion: unexpected state 'delete-failed', wanted target 'delete-complete'. last error: %!s(<nil>)
```

### Expected Behavior

Unassign prefix list, delete prefix list, create prefix list, assign prefix list

### Actual Behavior

Had to delete the association by hand from the security group and routing table

### Steps to Reproduce

Change the dynamic (max) entries to force a replacement of the pl.

**Comment by ewbankkit** [comment]
We need to use the `StateMessage` field in the [Describe API response](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_ManagedPrefixList.html) to populate that last error field.

**Comment by ewbankkit** [comment]
I added the additional error message in #19394 which should be available in Terraform AWS Provider **v3.60.0**.

**Comment by BornToBeRoot** [comment]
Hi @ewbankkit , would it make sense to add `name_prefix` as argument to the prefix list to force the creation of a new prefix list when something has changed?

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ec2_managed_prefix_list

Similiar to the security groups:

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/security_group#name_prefix
